sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionAsyncClient.java

```diff
@@ -0,0 +1,330 @@
+                        String token = httpRequest.getHeaders().getValue(HttpHeaderName.AUTHORIZATION);
+                        if (token != null && token.startsWith("Digest")) {
+                            MessageDigest messageDigest = null;
+                            try {
+                                messageDigest = MessageDigest.getInstance("SHA-256");
+                            } catch (NoSuchAlgorithmException e) {
+                                e.printStackTrace();
+                            }
+                            String authString = proxyOptions.getUsername() + ":" + proxyOptions.getPassword();
+                            messageDigest.update(authString.getBytes());
+                            String authStringEnc = Base64.getEncoder().encodeToString(messageDigest.digest());
+                            connection.setRequestProperty("Proxy-Authorization", "Digest " + authStringEnc);
+                        } else{
+                            String authString = proxyOptions.getUsername() + ":" + proxyOptions.getPassword();
+                            String authStringEnc = Base64.getEncoder().encodeToString(authString.getBytes());
+                            connection.setRequestProperty("Proxy-Authorization", "Basic " + authStringEnc);
+                        }
+                    }
+                }
+            } else {
+                connection = (HttpURLConnection) url.openConnection();
+            }
+
+            assert connection != null;
+
+            if (connectionTimeout != -1) {
+                connection.setConnectTimeout((int) connectionTimeout);
+            }
+
+            if (readTimeout != -1) {
+                connection.setReadTimeout((int) readTimeout);
+            }
+
+            // SetConnectionRequest
+            try {
+                connection.setRequestMethod(httpRequest.getHttpMethod().toString());
+            } catch (ProtocolException e) {
+                throw LOGGER.logExceptionAsError(new RuntimeException(e));
+            }
+
+            for (HttpHeader header : httpRequest.getHeaders()) {
+                for (String value : header.getValues()) {
+                    connection.addRequestProperty(header.getName(), value);
+                }
+            }
+            return connection;
+        } catch (IOException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+        }
+    }
+
+    /**
+     * Asynchronously sends the content of an HttpRequest via an HttpUrlConnection instance.
+     *
+     * @param httpRequest The HTTP Request being sent.
+     * @param progressReporter A reporter for the progress of the request.
+     * @param connection The HttpURLConnection that is being sent to.
+     * @return A Mono that represents the completion of the request sending process.
+     */
+    private Mono<Void> sendAsyncRequest(HttpRequest httpRequest, ProgressReporter progressReporter, HttpURLConnection connection) {
```

Comment:
Perhaps there exists prior art in other http client implementations, but this might be better named `sendBodyAsync`